### PR TITLE
Added cookie consent popup. Consent config needs to be edited.

### PR DESCRIPTION
**Pull request by samyak-max**
Screenshot (290)

The CookieConsentConfig.js file needs to be edited to add links for "privacy policy," "contact us," and config "performance and analytics cookies," and "advertisement and targeting cookies" in the consent settings.
Screenshot (291)

### CODE REVIEW

**Comment by tanya17-05** [comment]
LGTM 👍
@samyak-max thankyou for the contribution, can you please properly configure the remaining settings and remove all the unnecessary comments?

**Review by uniqueg — CHANGES_REQUESTED** [review]
LGTM, too, @samyak-max - thanks a lot :)

Agreeing with @tanya17-05 - please try to add the remaining config. I'm not actually sure the website uses any cookies at the moment (can you please tell us, @git-anurag-hub, or maybe you also know, @tanya17-05), so those may need to remain empty for now, at least until somebody addresses #52). For the privacy policy you can link to the `/policy` route, even if it's not yet existing (see #54).

@git-anurag-hub may have some more comments (e.g., on fonts/style used compared to the rest of the page etc)

File: components/CookieConsentConfig.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
+/**
```
Comment (git-anurag-hub) [inline components/CookieConsentConfig.js:1]: This file shouldn't be in the components folder. Move `pluginConfig` to the component it is used in.

File: components/CookieConsentConfig.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+  // cookie_same_site: 'Lax',                // default: 'Lax'
+  // use_rfc_cookie: false,                  // default: false
+  // revision: 0,                            // default: 0
+
+  onFirstAction: function (user_preferences, cookie) {
+    // callback triggered only once
+  },
+
+  onAccept: function (cookie) {
+    // ...
```
Comment (git-anurag-hub) [inline components/CookieConsentConfig.js:30]: Nothing happens when someone consents to the cookies!

File: components/CookieConsentConfig.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+  },
+
+  onChange: function (cookie, changed_preferences) {
+    // ...
```
Comment (git-anurag-hub) [inline components/CookieConsentConfig.js:34]: Nothing happens when someone change consents to the cookies!

**Review by git-anurag-hub — CHANGES_REQUESTED** [review]
Great Work @samyak-max, can you please implement the functionality to this component?
Right now it will just make the UI for the component but it doesn't do anything. You can get more info here. https://javascript.plainenglish.io/manage-cookie-consent-in-next-js-with-google-tag-manager-4d58818266ea

Reviewed for functionality for now. Will do UI changes once the functionality is implemented.

File: components/CookieConsentConfig.js (hunk 1 above)
Comment (samyak-max) [reply components/CookieConsentConfig.js:1]: Done ✅

File: components/CookieConsentConfig.js (hunk 2 above)
Comment (samyak-max) [reply components/CookieConsentConfig.js:30]: The cookies are stored whenever the Accept button is clicked; this function is for loading extra scripts like for the GA4 tag. 

Screenshot (291)

For GA4, the resource you shared, a Google Tag Manager account is to be made, which is linked to the google account so, should I do it with my personal account or is there a G-suit for that?

File: components/CookieConsentConfig.js (hunk 3 above)
Comment (samyak-max) [reply components/CookieConsentConfig.js:34]: This function is for some cleanup logic implementations like disabling the GA4 scripts (which is only valid if we add the user's functionality to change the consent after being accepted once as this function will be executed only if consent has already been given). The consent content is changing.

Screenshot (294)

Pls, let me know if I am missing anything 😓

**Comment by uniqueg** [comment]
@samyak-max: please do the changes as discussed with @git-anurag-hub on #79. Closing this one now

**Pull request closed without merging**